panda/src/navigation/navMesh.h

```diff
@@ -0,0 +1,101 @@
+/**
+ * PANDA 3D SOFTWARE
+ * Copyright (c) Carnegie Mellon University.  All rights reserved.
+ *
+ * All use of this software is subject to the terms of the revised BSD
+ * license.  You should have received a copy of this license along
+ * with this source code in a file named "LICENSE."
+ *
+ * @file navMesh.h
+ * @author ashwini
+ * @date 2020-060-21
+ */
+
+
+#ifndef NAVMESH_H
+#define NAVMESH_H
+
+#include "DetourNavMesh.h"
+#include "DetourNavMeshBuilder.h"
+#include "typedWritableReferenceCount.h"
+#include "pandaFramework.h"
+#include "pandaSystem.h"
+
+class EXPCL_NAVIGATION NavMeshParams {
+PUBLISHED:
+  int vert_count;
+  int poly_count;
+  int nvp;
+  int detail_vert_count;
+  int detail_tri_count;
+  float walkable_height;
+  float walkable_radius;
+  float walkable_climb;
+  float cs;
+  float ch;
+  bool build_bv_tree;
+  float b_min[3];
+  float b_max[3];
+  unsigned short *verts;        // size: 3 * vert_count
+  unsigned short *polys;        // size: poly_count * 2 * nvp
+  unsigned short *poly_flags;   // size: poly_count
+  unsigned char *poly_areas;    // size: poly_count
+  unsigned int *detail_meshes;  // size: poly_count * 4
+  float *detail_verts;          // size: detail_vert_count * 3
+  unsigned char *detail_tris;   // size: detail_tri_count * 4
+  int RC_MESH_NULL_IDX = 65535;
+};
+
+class EXPCL_NAVIGATION NavMesh: public TypedWritableReferenceCount
+{
+PUBLISHED:
+  NavMesh(dtNavMesh *nav_mesh);
+  NavMesh(NavMeshParams mesh_params);
+  void set_nav_mesh(dtNavMesh *m) { _nav_mesh = m; }
+  NavMesh();
+  PT(GeomNode) draw_nav_mesh_geom();
+  int get_vert_count() { return _params.vertCount; }
+  int get_poly_count() { return _params.polyCount; }
+  int get_detail_vert_count() { return _params.detailVertsCount; }
+  int get_detail_tri_count() { return _params.detailTriCount; }
+
+private:
+  dtNavMesh *_nav_mesh;
+  dtNavMeshCreateParams _params;
+  int RC_MESH_NULL_IDX;
```

Comment:
Again, since this is a constant, this should not be stored (and if it really should be stored, it should conform to the naming conventions for members).

Reply:
@rdb the vertices with this value belong to the edges which are solid borders. So if someone does not use navmeshgen to build the navmesh, then he/she might have a different value for such vertices.

Reply:
Done